kernel/src/engine/default/json.rs

```diff
@@ -88,6 +90,51 @@ impl<E: TaskExecutor> JsonHandler for DefaultJsonHandler<E> {
             self.readahead,
         )
     }
+
+    // note: for now we just buffer all the data and write it out all at once
+    fn write_json_file<'a>(
```

Comment:
```suggestion
    fn write_json_file(
        &self,
        path: &url::Url,
        data: Box<dyn Iterator<Item = Box<dyn EngineData>> + Send + '_>,
        _overwrite: bool,
    ) -> DeltaResult<()> {
```

Could just use the default lifetime.

Reply:
nice thanks, surprised clippy didn't find this?

Reply:
I don't know how deep it's lifetime analysis goes, but I checked out your branch to validate it before I made the suggestion.